### ISSUE TITLE
concurrent.futures.process.BrokenProcessPool: A process in the process pool was terminated abruptly while the future was running or pending.

### ISSUE THREAD

**Issue by Kang1448** [nipreps/fmriprep#2499]
### What version of fMRIPrep are you using?
20.2.1
### What kind of installation are you using? Containers (Singularity, Docker), or "bare-metal"?
Docker
### What is the exact command-line you used?
```
bash fmriprep.sh--low-mem
```

### Have you checked that your inputs are BIDS valid?
Yes
### Can you find some traces of the error reported in the visual report (at the bottom) or in *crashfiles*?
```
<please copy&paste all the information you can gather about errors>
```

### Are you reusing previously computed results (e.g., FreeSurfer, Anatomical derivatives, work directory of previous run)?
Nope 

### fMRIPrep log
If you have access to the output logged by fMRIPrep, please make sure to attach it as a text file to this issue.

[CITATION.md](https://github.com/nipreps/fmriprep/files/6957955/CITATION.md)

**Comment by effigies** [comment]
https://fmriprep.org/en/stable/faq.html#my-fmriprep-run-is-hanging

**Comment by Kang1448** [comment]
I have 12 CPUs and 16G RAM with "--low-mem," which I think is sufficient for fMRI prep. Why did it still return me the error? Do you know why?

**Comment by effigies** [comment]
It's difficult to say without more information. I don't know what `fmriprep.sh` contains. Docker often has less memory available to it than the full system. You can check whether that's true in your case. If the problem persists, you can try setting `--mem-mb 12000` to ensure that we leave a bigger buffer in case jobs take more memory than expected.

**Comment by Kang1448** [comment]
Here is the content of fmriprep.sh:
#!/bin/bash
#User inputs:
bids_root_dir=$HOME/learning/
subj=01
nthreads=16
mem=100 #gb
container=docker #docker or singularity

#Begin:

#Convert virtual memory from gb to mb
mem=`echo "${mem//[!0-9]/}"` #remove gb at end
mem_mb=`echo $(((mem*1000)-5000))` #reduce some memory for buffer space during pre-processing

export TEMPLATEFLOW_HOME=/Users/adm/Library/Python/3.7/lib/python/site-packages/templateflow/
export FS_LICENSE=$HOME/learning/derivatives/
cd $FS_LICENSE
ls

#Run fmriprep
if [ $container == singularity ]; then
  unset PYTHONPATH; singularity run -B $HOME/templateflow:/opt/templateflow /rds/general/user/wk20/home/fmriprep.simg \
    $bids_root_dir $bids_root_dir/derivatives \
    participant \
    --participant-label $subj \
    --md-only-boilerplate \
    --fs-license-file $HOME/learning/derivatives/license.txt \
    --fs-no-reconall \
    --output-spaces MNI152NLin2009cAsym:res-2 \
    --nthreads $nthreads \
    --stop-on-first-crash \
    --mem_mb $mem_mb \
    -w $HOME/learning/
else
  fmriprep-docker $bids_root_dir $bids_root_dir/derivatives \
    participant \
    --participant-label $subj \
    --skip-bids-validation \
    --md-only-boilerplate \
    --fs-license-file $HOME/learning/derivatives/license.txt \
    --fs-no-reconall \
    --output-spaces MNI152NLin2009cAsym:res-2 \
    --nthreads $nthreads \
    --stop-on-first-crash \
    --mem_mb $mem_mb \
    -w $HOME
fi

Do you mean I should try to run bash fmriprep.sh--mem-mb 12000?

**Comment by effigies** [comment]
I think the big problem is likely to be that you are telling fMRIPrep that it has 100GB instead of 16GB to work with.

> ```bash
> mem=100 #gb
> ```

You are also telling the script that you have 16 cores instead of 12. I would suggest as a first step just dropping `--nthreads` and `--mem_mb`. fMRIPrep tries to make reasonable first guesses. I also would not `--skip-bids-validation`, since that can detect issues in your data that are likely to cause errors in fMRIPrep.

As far as I can tell, adding arguments to `bash fmriprep.sh` isn't going to have any effect at all, since they're never referenced in the script.

**Comment by Kang1448** [comment]
<img width="920" alt="BD236E09-0BF8-4E86-8A7A-8B9CDD6E450C" src="https://user-images.githubusercontent.com/67905967/129214333-74373388-3836-4595-a081-224c450cf9bf.png">
<img width="390" alt="1BFFFAAC-034E-45D9-8918-9A6C6C733822" src="https://user-images.githubusercontent.com/67905967/129214335-fe19b97c-3e66-42c3-8756-74a51e585b60.png">
<img width="567" alt="CD0DCB19-7AB3-4A17-83CE-24622E3338B2" src="https://user-images.githubusercontent.com/67905967/129214334-3b435650-0787-46d2-839a-95ca9e4510ca.png">
<img width="516" alt="9431800F-E8F3-4070-A124-29F0F9BBA2CF" src="https://user-images.githubusercontent.com/67905967/129214343-979002ce-3bfc-4c7b-841f-d1ca0eb1ef76.png">

**Comment by Kang1448** [comment]
I did validate it on https://bids-standard.github.io/bids-validator/. And it returned what I sent you

File Path: Quick validation failed - the general folder structure does not resemble a BIDS dataset. Have you chosen the right folder (with "sub-*/" subfolders)? Check for structural/naming issues and presence of at least one subject.

Type:		Error
File:		sub-01
Reason:		Quick validation failed - the general folder structure does not resemble a BIDS dataset. Have you chosen the right folder (with "sub-*/" subfolders)? Check for structural/naming issues and presence of at least one subject.

**Comment by Kang1448** [comment]
Hello, 
May I know any update?

**Comment by effigies** [comment]
fMRIPrep needs a compliant BIDS dataset. You need to point it at a directory that validates. The best place for support regarding BIDS is https://neurostars.org.

**Comment by oesteban** [comment]
This is a duplicate of #2492 and #2491.

#2492 was closed in favor of #2491. #2491 was closed because we did not have enough information to help further. There are two more messages there, after closing the issue, but I still don't find the requested documentation. @Kang1448, if the problem is that we haven't successfully transmitted to you what is being requested, please let us know - we will be happy to assist and will make efforts to make our documentation clearer and our messages accurate and effective. Otherwise, we will have to assume that you are not getting satisfactory answers because you are failing to read fMRIPrep's documentation and how we provide support for it. We have very limited bandwidth to attend to support requests, and we need to efficiently and equally divide that time across all of our users.

In any case, please follow up on the original thread - which is #2491.
